hw-35/script.ts

```diff
@@ -0,0 +1,102 @@
+// С ниже приведенным массивом решить следующие задачи. Все функции и данные должны быть протипизированы:
+
+interface IUser {
+    name: string; 
+    phone: string;
+    email: string; 
+    animals?: string[]; 
+    cars?: string[]; 
+    hasChildren: boolean; 
+    hasEducation: boolean;
+}
+
+const users: Array<IUser> = [
+    {
+        name: "Harry Felton",
+        phone: "(09) 897 33 33",
+        email: "felton@gmail.com",
+        animals: ["cat"],
+        cars: ["bmw"],
+        hasChildren: false,
+        hasEducation: true    
+    },
+    {
+        name: "May Sender",
+        phone: "(09) 117 33 33",
+        email: "sender22@gmail.com",
+        hasChildren: true,
+        hasEducation: true
+    },
+    {
+        name: "Henry Ford",
+        phone: "(09) 999 93 23",
+        email: "ford0@gmail.com",
+        cars: ["bmw", "audi"],
+        hasChildren: true,
+        hasEducation: false
+    }
+]
+
+
+// 1. Создать строку из имен пользователей через запятую
+function getUsersName(users: Array<IUser>):string {
+    return users.map(({name}) => name).join(', ');
+}
+console.log(getUsersName(users));
+
+
+// 2. Посчитать общее количество машин у пользователей
+function getUsersCars(users: Array<IUser>):number {
+    let i = 0;
+    users.map(({cars}) => {
+        if (cars) {
```

Comment:
You can avoid this condition by using ternary operator:

`users.map(({cars}) => cars ? cars.map(() => i++) : [])`